pkg/block/azure/client_cache.go

```diff
@@ -0,0 +1,91 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+	"github.com/puzpuzpuz/xsync"
+	"github.com/treeverse/lakefs/pkg/block/params"
+)
+
+type ClientContainerCache struct {
+	containerToClient *xsync.MapOf[string, *container.Client]
+	params            params.Azure
+}
+
+var ErrUnknownAzureURL = errors.New("unable to parse storage account")
+
+func NewCache(p params.Azure) *ClientContainerCache {
+	return &ClientContainerCache{
+		containerToClient: xsync.NewMapOf[*container.Client](),
+		params:            p,
+	}
+}
+
+// storageAccountRegex is used to extract the storage account from azure storage url.
+// Following rules from https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview#storage-account-name
+var storageAccountRegex = regexp.MustCompile("https://([a-z0-9]{3,24}).blob.core.windows.net/.*")
+
+func ExtractStorageAccount(storageAccountURL string) (string, error) {
```

Comment:
This can be apart of the BlobURLInfo or PrefixURLInfo and extracted usng the ResolveBlobURLInfoFromURL.
I don't think we need a special function just for the storage account

Reply:
Done. The function is still needed for the walker flow in ingest.